crates/hir-ty/src/mir/eval/shim.rs

```diff
@@ -309,43 +334,6 @@ impl Evaluator<'_> {
         let mut args = args.iter();
         match it {
             BeginPanic => Err(MirEvalError::Panic("<unknown-panic-payload>".to_owned())),
-            PanicFmt => {
```

Comment:
Don't we need to have something like this? How `const_panic_fmt` can return `MirEvalError::Panic` on its own?

Reply:
https://github.com/rust-lang/rust/blob/6a92312a1eaf5d9cffe3bdf9aaecbfdf8e2e35d9/library/core/src/panicking.rs#L259
it calls `panic_display` which is in turn hooked again https://github.com/rust-lang/rust/blob/6a92312a1eaf5d9cffe3bdf9aaecbfdf8e2e35d9/library/core/src/panicking.rs#L190

Reply:
So we need to hook `panic_display` in r-a as well? What is the output of `rust-analyzer(debug-command) interpret function` on a function that panics? It should be `MirEvalError::Panic("panic message")` but I guess it is now `execution limit exceeded` or something like that as it will try to execute `panic_fmt` and `panic_display` over and over.

Reply:
Hooking that function is what my `rustc_const_panic_str` change does.

Reply:
Ah I see. So we now lose the panic message? but let's track it in #16938 and merge this one.